api/src/mutations/drop.rs

```diff
@@ -176,6 +173,98 @@ impl Mutation {
         })
     }
 
+    /// This mutation retries an existing drop.
+    /// The drop returns immediately with a creation status of CREATING.
+    /// You can [set up a webhook](https://docs.holaplex.dev/hub/For%20Developers/webhooks-overview) to receive a notification when the drop is ready to be minted.
+    /// Errors
+    /// The mutation will fail if the drop and its related collection cannot be located,
+    /// if the transaction response cannot be built,
+    /// or if the transaction event cannot be emitted.
+    pub async fn retry_drop(
+        &self,
+        ctx: &Context<'_>,
+        input: RetryDropInput,
+    ) -> Result<CreateDropPayload> {
+        let AppContext {
+            db,
+            user_id,
+            organization_id,
+            balance,
+            ..
+        } = ctx.data::<AppContext>()?;
+        let UserID(id) = user_id;
+        let OrganizationId(org) = organization_id;
+        let producer = ctx.data::<Producer<NftEvents>>()?;
+        let solana = ctx.data::<Solana>()?;
+        let credits = ctx.data::<CreditsClient<Actions>>()?;
+        let user_id = id.ok_or_else(|| Error::new("X-USER-ID header not found"))?;
+        let org_id = org.ok_or_else(|| Error::new("X-ORGANIZATION-ID header not found"))?;
+        let balance = balance
+            .0
+            .ok_or_else(|| Error::new("X-ORGANIZATION-BALANCE header not found"))?;
+        let (drop, collection) = Drops::find_by_id(input.drop)
+            .find_also_related(Collections)
+            .one(db.get())
+            .await?
+            .ok_or_else(|| Error::new("drop not found"))?;
+
+        let collection = collection.ok_or_else(|| Error::new("collection not found"))?;
+
+        if drop.creation_status == CreationStatus::Created {
+            return Err(Error::new("drop already created"));
+        }
+
+        let (
+            collection_address,
+            TransactionResponse {
+                serialized_message,
+                signed_message_signatures,
+            },
+        ) = match collection.blockchain {
+            BlockchainEnum::Solana => {
+                solana
+                    .edition()
+                    .retry_drop(RetryDropRequest {
+                        collection: collection.clone(),
+                    })
+                    .await?
```

Comment:
For credits retrying a drop or mint should confirm the credit deduction for the original request and not worry about charging for the gas required for the retry.